File: practica1.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        
        System.out.println("¿Qué quieres hacer?");
        System.out.println("1) Dividir");
        System.out.println("2) Multiplicar");
        System.out.println("3) Calcular el factorial de un número");
        
        int opcion = input.nextInt();
        
        if (opcion == 1) {
            System.out.print("Ingresa el dividendo: ");
            int dividendo = input.nextInt();
            System.out.print("Ingresa el divisor: ");
            int divisor = input.nextInt();
            
            int[] resultado = divisionPorRestas(dividendo, divisor);
            System.out.println("Cociente: " + resultado[0]);
            System.out.println("Residuo: " + resultado[1]);
        } else if (opcion == 2) {
            System.out.print("Ingresa el primer número: ");
            int numero1 = input.nextInt();
            System.out.print("Ingresa el segundo número: ");
            int numero2 = input.nextInt();
            
            int resultado = multiplicacionPorSumas(numero1, numero2);
            System.out.println("El resultado de " + numero1 + " x " + numero2 + " es: " + resultado);
        } else if (opcion == 3) {
            System.out.print("Ingresa un número para calcular su factorial: ");
            int numero = input.nextInt();
            
            int resultado = factorial(numero);
            System.out.println("El factorial de " + numero + " es: " + resultado);
        } else {
            System.out.println("Opción incorrecta");
        }
        
        input.close();
    }
    
    public static int[] divisionPorRestas(int numero, int divisor) {
        int cociente = 0;
        
        while (numero >= divisor) {
            numero -= divisor;
            cociente++;
        }
        
        int[] resultado = {cociente, numero};
        return resultado;
    }
    
    public static int multiplicacionPorSumas(int numero1, int numero2) {
        int resultado = 0;
        
        for (int i = 0; i < numero2; i++) {
            resultado += numero1;
        }
        
        return resultado;
    }
    
    public static int factorial(int numero) {
        if (numero == 0) {
            return 1;
        } else {
            return numero * factorial(numero - 1);
        }
    }
}

File: practica1.cpp
#include <iostream>

using namespace std;

int divisionPorRestas(int numero, int divisor) {
    int cociente = 0;
    
    while (numero >= divisor) {
        numero -= divisor;
        cociente++;
    }
    
    return cociente;
}

int multiplicacionPorSumas(int numero1, int numero2) {
    int resultado = 0;
    
    for (int i = 0; i < numero2; i++) {
        resultado += numero1;
    }
    
    return resultado;
}

int factorial(int numero) {
    if (numero == 0) {
        return 1;
    } else {
        return numero * factorial(numero - 1);
    }
}

int main() {
    int opcion;
    cout << "Que quieres hacer?" << endl;
    cout << "1) Dividir" << endl;
    cout << "2) Multiplicar" << endl;
    cout << "3) Calcular el factorial de un numero" << endl;
    
    cin >> opcion;
    
    if (opcion == 1) {
        int dividendo, divisor;
        cout << "Ingresa el dividendo: ";
        cin >> dividendo;
        cout << "Ingresa el divisor: ";
        cin >> divisor;
        
        int cociente = divisionPorRestas(dividendo, divisor);
        int residuo = dividendo - cociente * divisor;
        
        cout << "Cociente: " << cociente << endl;
        cout << "Residuo: " << residuo << endl;
    } else if (opcion == 2) {
        int numero1, numero2;
        cout << "Ingresa el primer número: ";
        cin >> numero1;
        cout << "Ingresa el segundo número: ";
        cin >> numero2;
        
        int resultado = multiplicacionPorSumas(numero1, numero2);
        
        cout << "El resultado de " << numero1 << " x " << numero2 << " es: " << resultado << endl;
    } else if (opcion == 3) {
        int numero;
        cout << "Ingresa un número para calcular su factorial: ";
        cin >> numero;
        
        int resultado = factorial(numero);
        
        cout << "El factorial de " << numero << " es: " << resultado << endl;
    } else {
        cout << "Opción incorrecta" << endl;
    }
    
    return 0;
}
